File: TransCoder_tokenized_test_set_functions/cpp/FIND_POSITION_GIVEN_NUMBER_AMONG_NUMBERS_MADE_4_7.cpp
int findpos ( string n ) {
  int i = 0, pos = 0;
  while ( n [ i ] != '\0' ) {
    switch ( n [ i ] ) {
      case '4' : pos = pos * 2 + 1;
      break;
      case '7' : pos = pos * 2 + 2;
      break;
    }
    i ++;
  }
  return pos;
}

File: TransCoder_tokenized_test_set_functions/java/FIND_POSITION_GIVEN_NUMBER_AMONG_NUMBERS_MADE_4_7.java
static int findpos ( String n ) {
  int k = 0, pos = 0, i = 0;
  while ( k != n . length ( ) ) {
    switch ( n . charAt ( i ) ) {
      case '4' : pos = pos * 2 + 1;
      break;
      case '7' : pos = pos * 2 + 2;
      break;
    }
    i ++;
    k ++;
  }
  return pos;
}